core/dbt/contracts/graph/manifest.py

```diff
@@ -1020,26 +1023,30 @@ def build_group_map(self):
                     group_map[node.group].append(node.unique_id)
         self.group_map = group_map
 
+    @classmethod
+    def _map_nodes_to_map_resources(cls, nodes_map: MutableMapping[str, NodeClassT]):
+        return {node_id: node.to_resource() for node_id, node in nodes_map.items()}
+
     def writable_manifest(self) -> "WritableManifest":
         self.build_parent_and_child_maps()
         self.build_group_map()
         return WritableManifest(
             nodes=self.nodes,
-            sources=self.sources,
+            sources=self._map_nodes_to_map_resources(self.sources),
             macros=self.macros,
             docs=self.docs,
-            exposures=self.exposures,
-            metrics=self.metrics,
+            exposures=self._map_nodes_to_map_resources(self.exposures),
+            metrics=self._map_nodes_to_map_resources(self.metrics),
             groups=self.groups,
```

Comment:
I believe we should also implement the resource_class method on Group in nodes.py and use self._map_nodes_to_map_resources on it here

Reply:
Good catch!